src/SignalR.Orleans/OrleansHubLifetimeManager.cs

```diff
@@ -145,15 +138,27 @@ public override Task InvokeGroupAsync(string groupName, string methodName, objec
 
         public override Task InvokeUserAsync(string userId, string methodName, object[] args)
         {
-            // TODO: Check with @davidfowl what is this method for
-            throw new NotImplementedException();
+            if (string.IsNullOrWhiteSpace(userId)) throw new ArgumentNullException(nameof(userId));
+            if (string.IsNullOrWhiteSpace(methodName)) throw new ArgumentNullException(nameof(methodName));
+
+            var message = new InvocationMessage(Guid.NewGuid().ToString(), nonBlocking: true, target: methodName, arguments: args);
+            var user = this._clusterClient.GetGrain<IUserGrain>(userId);
+            return user.SendMessage(message);
         }
 
         public override async Task OnConnectedAsync(HubConnectionContext connection)
         {
             try
             {
                 this._connections.Add(connection);
+
+                if (connection.User.Identity.IsAuthenticated)
+                {
+                    //TODO: replace `connection.User.Identity.Name` with `connection.UserIdentifier` when next signalr will be published.

```

Comment:
Where did you came up that we should replace it? (just curious)

Reply:
I was diving in the SignalR code and I found noticed that they updated the `DefaultHubLifetimeManager` with `connection.UserIdentifier`.  This will allow the consumers to override the `UserIdentifier` so it can be configured instead of claim name.

https://github.com/aspnet/SignalR/blob/cb3124be17c8d7f476e2f4eab3c1c9fbe213f23d/src/Microsoft.AspNetCore.SignalR.Core/HubEndPoint.cs#L78

 https://github.com/aspnet/SignalR/blob/665f166d67bedb6df53499ab59b618e8a72cf2c0/src/Microsoft.AspNetCore.SignalR.Core/DefaultUserIdProvider.cs

https://github.com/aspnet/SignalR/blob/cb3124be17c8d7f476e2f4eab3c1c9fbe213f23d/src/Microsoft.AspNetCore.SignalR.Core/DefaultHubLifetimeManager.cs#L134

https://docs.microsoft.com/en-us/aspnet/signalr/overview/guide-to-the-api/mapping-users-to-connections#IUserIdProvider